src/tasks/models/tasksManager.ts

```diff
@@ -0,0 +1,130 @@
+  //   const jobs = await this.jobManagerClient.getJobsStatus();
+  //   const completedJobs = jobs?.filter((job) => (job.completedTasks === job.taskCount) || (job.failedTasks === job.taskCount));
+  //   return completedJobs!;
+  // }
+
+  public async getTaskStatusByJobId(jobId: string): Promise<ITaskStatusResponse> {
+    this.logger.info(`Getting task status by jobId: ${jobId}`);
+    const tasks = await this.jobManagerClient.getTasksByJobId(jobId);
+
+    if (tasks.length === 0) {
+      throw new NotFoundError(`jobId: ${jobId} is not exists`);
+    }
+    const task = tasks[0];
+    const statusResponse: ITaskStatusResponse = {
+      percentage: task.percentage,
+      status: task.status,
+    };
+    return statusResponse;
+  }
+
+  public async sendCallbacks(job: JobResponse, expirationDate: Date, errorReason?: string): Promise<ICallbackDataBase | undefined> {
+    try {
+      const downloadServerUrl = config.get('downloadServerUrl');
+      const packageName = job.parameters.fileName as string;
+      const fileUri = `${downloadServerUrl}/downloads/${packageName}`;
+      const packageFullPath = getGpkgFilePath(this.gpkgsLocation, packageName);
+      const success = errorReason === undefined;
+      let fileSize = 0;
+      if (success) {
+        fileSize = await getFileSize(packageFullPath);
+      }
+      const callbackParams: ICallbackDataBase = {
+        fileUri,
+        expirationTime: expirationDate,
+        fileSize,
+        dbId: job.internalId as string,
+        packageName: packageName,
+        requestId: job.id,
+        targetResolution: job.parameters.targetResolution,
+        success,
+        errorReason,
+      };
+
+      const targetCallbacks = job.parameters.callbacks;
+      const callbackPromises: Promise<void>[] = [];
+      for (const target of targetCallbacks) {
+        const params: ICallbackData = { ...callbackParams, bbox: target.bbox };
+        callbackPromises.push(this.callbackClient.send(target.url, params));
+      }
+
+      const promisesResponse = await Promise.allSettled(callbackPromises);
+      promisesResponse.forEach((response, index) => {
+        if (response.status === 'rejected') {
+          this.logger.error(`Did not send callback to ${targetCallbacks[index].url}, got error: ${JSON.stringify(response.reason)}`);
+        }
+      });
+      return callbackParams;
+    } catch (error) {
+      this.logger.error(`Sending callbacks has failed with error: ${JSON.stringify(error, Object.getOwnPropertyNames(error))}`);
+    }
+  }
+
+  public async finalizeJob(job: JobResponse, expirationDate: Date, isSuccess = true, reason?: string): Promise<void> {
+    let updateJobParams: IUpdateJobBody<IJobParameters> = {
+      status: isSuccess ? OperationStatus.COMPLETED : OperationStatus.FAILED,
+      reason,
+      percentage: isSuccess ? 100 : undefined,
+      expirationDate: expirationDate,
+    };
+    try {
+      if (isSuccess) {
```

Comment:
callbacks needs to be called also on failure

Reply:
fixed